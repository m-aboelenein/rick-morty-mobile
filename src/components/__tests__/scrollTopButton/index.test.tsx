import {fireEvent, render} from '@testing-library/react-native';
import * as React from 'react';
import renderer from 'react-test-renderer';
import ScrollTopButton from '../../scrollTopButton';

const mockOnPressScroll = jest.fn();

it('renders a button that scrolls up', () => {
  const tree = renderer.create(
    <ScrollTopButton onPressScroll={mockOnPressScroll} />,
  );
  expect(tree).toMatchSnapshot();
});

it('calls the scroll handler onPress', () => {
  const {getByTestId} = render(
    <ScrollTopButton onPressScroll={mockOnPressScroll} />,
  );

  fireEvent.press(getByTestId('scroll-button'));
  expect(mockOnPressScroll).toHaveBeenCalled();
});
