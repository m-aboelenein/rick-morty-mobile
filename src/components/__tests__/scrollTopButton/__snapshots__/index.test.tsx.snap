// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a button that scrolls up 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "borderRadius": 20,
      "bottom": 60,
      "height": 40,
      "justifyContent": "center",
      "position": "absolute",
      "right": 30,
      "width": 40,
    }
  }
  testID="scroll-button"
>
  <svg
    data-file-name="SvgChevronLeft"
    height={30}
    style={
      Object {
        "transform": Array [
          Object {
            "rotate": "90deg",
          },
        ],
      }
    }
    width={30}
  />
</View>
`;
